wsd/DocumentBroker.cpp

```diff
@@ -1691,6 +1691,68 @@ void PresetsInstallTask::install(const Poco::JSON::Object::Ptr& settings,
     }
 }
 
+/*
+These functions are helpers for backward compatibility
+In older versions of Richdocument, Zotero API keys were handled by RD
+In newer versions of RD and COOL(25.04.5), the Zotero key is saved in AdminIntegratorSettings
+The following functions help to transfer the Zotero key from RD to COOL without user action
+Without this, the user will have to re-enter the api key in settings or Zotero settings
+will disappear, making the user think functionality is broken
+*/
+void uploadRDZoteroAPIKey(const std::shared_ptr<ClientSession>& session, std::string& zoteroAPIKey,
+                          const Poco::JSON::Object::Ptr& viewsetting,
+                          std::string& viewSettingsString)
+{
+    const_cast<Poco::JSON::Object::Ptr&>(viewsetting)->set("zoteroAPIKey", zoteroAPIKey);
+
+    Poco::URI wopiUri = DocumentBroker::getPresetUploadBaseUrl(session->getPublicUri());
+    if (wopiUri.getPath().starts_with("/index.php"))
+        wopiUri.setPath(wopiUri.getPath().substr(10));
+    wopiUri.addQueryParameter("fileId", "/settings/userconfig/viewsetting/viewsetting.json");
+    std::cout << wopiUri.toString() << std::endl;
+    const std::string& uriAnonym = COOLWSD::anonymizeUrl(wopiUri.toString());
+
+    auto httpRequest =
+        StorageConnectionManager::createHttpRequest(wopiUri, session->getAuthorization());
+    httpRequest.setVerb(http::Request::VERB_POST);
+
+    httpRequest.set("Content-Type", "application/octet-stream");
+
+    viewSettingsString = JsonUtil::jsonToString(viewsetting);
+    httpRequest.setBody(viewSettingsString);
+
+    auto httpSession = StorageConnectionManager::getHttpSession(wopiUri);
+    httpSession->asyncRequest(httpRequest, COOLWSD::getWebServerPoll());
+}
+
+void fetchAndReuseRDZoteroAPIKey(const std::shared_ptr<ClientSession>& session,
+                                 std::string& zoteroAPIKey,
+                                 const Poco::JSON::Object::Ptr& viewsetting,
+                                 std::string& viewSettingsString)
+{
+    const std::string& userPrivateInfo = session->getUserPrivateInfo();
+    Object::Ptr userPrivateInfoObj;
+    Poco::JSON::Parser parser;
+    Poco::Dynamic::Var var = parser.parse(userPrivateInfo);
+    try
+    {
+        userPrivateInfoObj = var.extract<Object::Ptr>();
+    }
+    catch (const Poco::BadCastException& exception)
```

Comment:
I think move the .parse call inside the try/catch block and catch const std::exception& (like I see in extractViewSettings) to catch anything that might happen so there's no risk of crashing the entire coolwsd on some unusual json